src/Datadog.Trace/AppSec/Waf/Waf.cs

```diff
@@ -66,53 +78,111 @@ public void Dispose(bool disposing)
-                var ruleHandle = WafNative.Init(rules.RawPtr, ref args);
+        private static WafHandle CreateWafHandle(string rulesFile)
+        {
+            var argCache = new List<Obj>();
+            Obj configObj;
+            try
+            {
+                var stream = GetRulesStream(rulesFile);
 
-                rules.Dispose();
-                foreach (var arg in argCache)
+                if (stream == null)
                 {
-                    arg.Dispose();
+                    return null;
                 }
 
-                if (ruleHandle == IntPtr.Zero)
+                if (Log.IsEnabled(LogEventLevel.Debug))
                 {
-                    Log.Error("Failed to create rules.");
-                    return null;
+                    LogRuleDetails(rulesFile);
                 }
-                else
+
+                configObj = CreatObjFromRulesStream(argCache, stream);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, $"AppSec could not read the rule file \"{rulesFile}\" as it was invalid. AppSec will not run any protections in this application.");
+                return null;
+            }
+
+            try
+            {
+                DdwafConfigStruct args = default;
+                var ruleHandle = WafNative.Init(configObj.RawPtr, ref args);
+                return new WafHandle(ruleHandle);
+            }
+            finally
+            {
+                configObj?.Dispose();
+                foreach (var arg in argCache)
                 {
-                    Log.Information("Rules successfully created.");
+                    arg.Dispose();
                 }
+            }
+        }
 
-                return new WafHandle(ruleHandle);
+        private static void LogRuleDetails(string rulesFile)
+        {
+            try
+            {
+                // because of the generic way rules are encoded for the WAF there's no good
+                // way to log them without double parsing, but as this only happens at debug
+                // log level, I think this is okay
+                var stream = GetRulesStream(rulesFile);
+                var root = GetJsonReaderFromStream(stream);
+
+                var eventsProp = root.Value<JArray>("events");
+                foreach (var ev in eventsProp)
+                {
+                    var idProp = ev.Value<JValue>("id");
+                    var nameProp = ev.Value<JValue>("name");
+                    var addresses = ev.Value<JArray>("conditions").SelectMany(x => x.Value<JObject>("parameters").Value<JArray>("inputs"));
+                    Log.Debug($"Loaded rule: {idProp.Value} - {nameProp.Value} on addresses: {string.Join(", ", addresses)}");
+                }
             }
             catch (Exception ex)
             {
-                Log.Error(ex, "Error loading the power WAF rules");
-                return null;
+                Log.Error(ex, "Error occured logging rules");
```

Comment:
```suggestion
                Log.Error(ex, "Error occured logging the power WAF rules");
```

Reply:
we're trying to drop the "power waf" name in favour of ddwaf, but I add precision in 018555bcbbd99651306a9f7a6d3913a05b86bc6a